builtin/game/falling.lua

```diff
@@ -258,17 +258,41 @@ end
 -- Global callbacks
 --
 
-function on_placenode(p, node)
-	nodeupdate(p)
+local function on_placenode(p, node)
+	check_for_falling(p)
 end
 core.register_on_placenode(on_placenode)
 
-function on_dignode(p, node)
-	nodeupdate(p)
+local function on_dignode(p, node)
+	check_for_falling(p)
 end
 core.register_on_dignode(on_dignode)
 
-function on_punchnode(p, node)
-	nodeupdate(p)
+local function on_punchnode(p, node)
+	check_for_falling(p)
 end
 core.register_on_punchnode(on_punchnode)
+
+--
+-- Globally exported functions
+--
+
+-- TODO remove this function after the 0.4.15 release
+function nodeupdate(p)
+	core.log("deprecated", "nodeupdate: deprecated, please use core.check_for_falling instead")
+	check_for_falling(p)
+end
+
+-- TODO remove this function after the 0.4.15 release
+function nodeupdate_single(p)
+	core.log("deprecated", "nodeupdate_single: deprecated, please use core.check_single_for_falling instead")
+	check_single_for_falling(p)
+end
+
+function core.check_for_falling(p)
```

Comment:
Is doing this a Lua optimisation? If so should we be doing this everywhere from now on? and maybe edit builtin to do this throughout in another PR?